actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -107,6 +107,7 @@ def redirect_to(options = {}, response_options = {})
       allow_other_host = response_options.delete(:allow_other_host) { _allow_other_host }
 
       self.status = _extract_redirect_to_status(options, response_options)
+      ActionController.deprecator.warn("Passing `redirect_to` a status other than a valid 3XX redirect is deprecated and will be removed in a future version of Rails") unless redirect?
```

Comment:
```suggestion
    unless redirect?
      ActionController.deprecator.warn <<~MSG
        Passing a non-3XX status to `redirect_to` is deprecated and will be removed in Rails 8.1.
      MSG
    end
```

Indentation looks off, but something like this, maybe.

Reply:
I think we want to raise in the future, not remove / no-op.

Reply:
of course! thanks, I got a little mixed up looking at other examples of `deprecator.warn`

thanks for the support @zzak! I've updated the warning message 🚀